### DISCUSSION TITLE
wizard step sharing from model

### DISCUSSION THREAD

**Discussion by CushtyDev** [filamentphp/filament#14035] [Help]
### Package

Form builder

### Package Version

3.2.0

### How can we help you?

Hi, I am trying to share the wizard steps from my Model, but it's causing the layout to not be full width and the has skippable steps stops working, Can you shed some light on how to properly set up the wizard to share steps? Thanks 

```
   public function form(Form $form): Form
    {
        return $form->schema([
            Wizard::make(Application::formWizard())
        ]);

    }


    public function hasSkippableSteps(): bool
    {
        return true;
    }
```

```
public static function formWizard(): array
    {
        return [
            Step::make('Personal Information)
                ->description('Provide personal details')
                ->schema([
                    Select::make('user_id')
                        ->label('User')
                        ->options(User::all()->pluck('name', 'id'))
                        ->required(),
                    TextInput::make('first_name')
                        ->required(),
                    TextInput::make('last_name')
                        ->required(),
                    TextInput::make('public_profile_name')
                        ->required(),
```

it works in panel correctly if I use this 

```
 protected function getSteps(): array
    {
        return Application::formWizard();

    }
```

but I cant use getSteps in a livewire component, thanks

**Comment by e2d2-dev** [comment]
Hi,

just wrap the steps in a function

`
public static function firstStep(): Step
{
    return Step::make();
}
`

and add it to the Wizard

`
Wizard::make()->schema([
    WizardStep::firstStep(),
    WizardStep::secondStep(),
])
`
